### ISSUE TITLE
next_cell_input = torch.cat((input, hidden, target[i, :, :]), dim=1)

### ISSUE THREAD

**Issue by 13121283123** [jjdabr/forecastNet#10]
Thank you very much for answering my first question. I'm sorry, I have one more question.
For the first input X, we feed it to the first hidden block to provide the first predicted value yt+1. When a second predictive value yt+2 is provided, the input includes not only the second input X, but also the first predictive value yt+1 and the first hidden block. This is what I got after reading your article. Is my understanding correct？But the input value of the next hidden block in your code seems to be the input value X, the previous hidden block, and the real value of the first hidden block, target[i, :, :]) ， not the predicted value. How should I understand it?
 if is_training:
         next_cell_input = torch.cat((input, hidden, target[i, :, :]), dim=1)
 else:
        next_cell_input = torch.cat((input, hidden, outputs[i, :, :]), dim=1)

**Comment by jjdabr** [comment]
The target (ground truth) rather than the output (prediction) is fed to the next layer only during training. This is commonly referred to as "teacher forcing". You don't have to use it if you prefer not to, or you can randomly switch between providing the target and the output such as in the example here: https://pytorch.org/tutorials/intermediate/seq2seq_translation_tutorial.html#training-the-model.
